### PR TITLE
feat(core,schemas): log register

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Log register

## Linear Issue Reference

LOG-1512

## Testing

Pass all existing tests.
image

### CODE REVIEW

**Comment by IceHe** [comment]
Check session before sign-in and register

File: packages/core/src/routes/session.ts [hunk 1]
```diff
@@ -318,7 +318,12 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
       }),
     }),
     async (ctx, next) => {
+      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
+      ctx.log.sessionId = jti;
+
       const { username, password } = ctx.guard.body;
+      ctx.log.type = LogType.RegisterUsernamePassword;
+      ctx.log.username = username;
```
Comment (simeng-li) [inline packages/core/src/routes/session.ts:326]: Can we infer the rest of the log properties by giving a specific LogType? Looks like there is no property guard or type hint so far.

Comment (IceHe) [reply packages/core/src/routes/session.ts:326]: Added the linear issue LOG-2200

File: packages/core/src/routes/session.ts [hunk 2]
```diff
@@ -336,6 +341,8 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
       );
 
       const id = await generateUserId();
+      ctx.log.userId = id;
```
Comment (demonzoo) [inline packages/core/src/routes/session.ts:344]: nit: No need to declare `id`, just simply
```typescript
ctx.log.userId = await generateUserId();
```

**Review by demonzoo — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/session.ts (hunk 2 above)
Comment (IceHe) [reply packages/core/src/routes/session.ts:344]: cannot inline because there is another usage of `id`

**Comment by IceHe** [comment]
Rebased #656

**Comment by IceHe** [comment]
Rebased master to resolve conflicts.

**Pull request merged**